src/main/java/org/parabot/environment/api/utils/Time.java

```diff
@@ -53,6 +53,30 @@ public static boolean sleep(SleepCondition conn, int timeout) {
         return true;
     }
 
+    /**
+     * Sleeps until SleepCondition is valid, but with a minimum timeout.
+     *
+     * @param conn              the condition.
+     * @param timeout           the time in milliseconds before it stops sleeping.
+     * @param minimumTimeout    the minimum time to sleep.
+     *
+     * @return whether it ran successfully without timing out.
+     */
+    public static boolean sleep(SleepCondition conn, int timeout, int minimumTimeout) {
+        long start = System.currentTimeMillis();
```

Comment:
Why don't you call the sleep method on line 45 instead of replicating this code?

Reply:
Good point, I just completely forgot to.
It would look something like this:
```
public static boolean sleep(SleepCondition conn, int timeout, int minimumTimeout) {
        long start = System.currentTimeMillis();
        if(!sleep(conn, timeout)) return false;
        long t;
        if((t = System.currentTimeMillis() - start) < minimumTimeout) {
            Time.sleep((int)(minimumTimeout - t));
        }
        return true;
}
```
Haven't tested it yet, I'll test and change it asap if you want.

Reply:
I'd go for the first one, as that avoids the creation of a new variable if you just call `Sleep(SleepCondition, int)`